operator/v1/types_network.go

```diff
@@ -96,12 +96,32 @@ type DefaultNetworkDefinition struct {
 	OVNKubernetesConfig *OVNKubernetesConfig `json:"ovnKubernetesConfig,omitempty"`
 }
 
-// AdditionalNetworkDefinition configures an extra network that is available but not
+// MacVlanConfig contains configurations for macvlan interface.
+type MacVlanConfig struct {
+
+	// Host interface to enslave interface.
+	Master string `json:"master"`
+
+	// Ipam indicates which IPAM module will be used for IP Address Management(IPAM)
+	// Ipam for now supports 'dhcp'
+	// +optional
+	IPAM string `json:"ipam,omitempty"`
+
+	// Mode is the macvlan mode: bridge, private, vepa, passthru. The default is bridge
+	// +optional
+	Mode string `json:"mode,omitempty"`
```

Comment:
If there is a fixed set of allowed values then this should be done as `type MacvlanMode string` and a set of predefined `const` values, like with `SDNMode`

Reply:
got it. introduce MACVLANMode at new diff.